core/modules/ccontrol/testCControl.cc

```diff
@@ -950,6 +950,14 @@ static const std::vector<Antlr4CompareQueries> ANTLR4_COMPARE_QUERIES = {
         },
         "SELECT * FROM Filter WHERE NOT(filterId>1 AND filterId<6)"
     ),
+
+    // tests XOR
+    Antlr4CompareQueries(
+        "select ra_PS, objectId from Object where ra_PS > 1 XOR ra_PS < 2;",
+        "select ra_PS, objectId from Object where (ra_PS > 1 and not ra_PS < 2) OR (not ra_PS > 1 and ra_PS < 2)",
+        [](query::SelectStmt::Ptr const & selectStatement) {},
+        "SELECT ra_PS,objectId FROM Object WHERE (ra_PS>1 AND NOT ra_PS<2) OR (NOT ra_PS>1 AND ra_PS<2)"
+    ),
```

Comment:
With that nontrivial transforms I'd prefer to see more tests with XOR combinations, e.g.

    A XOR NOT B
    NOT A XOR NOT B
    A XOR B XOR C
    A XOR B OR C XOR D
    A XOR B AND C XOR D